crates/api_common/src/utils.rs

```diff
@@ -1000,6 +1008,56 @@ pub fn send_webmention(post: Post, community: &Community) {
   };
 }
 
+pub fn community_follower_state(community: &Community) -> CommunityFollowerState {
+  if community.local {
+    // Local follow is accepted immediately
+    CommunityFollowerState::Accepted
+  } else if community.visibility == CommunityVisibility::Private {
+    // Private communities require manual approval
+    CommunityFollowerState::ApprovalRequired
+  } else {
+    // remote follow needs to be federated first
+    CommunityFollowerState::Pending
+  }
+}
+
+pub async fn community_follow_many(
+  person: &Person,
+  to_follow: &Vec<Community>,
+  context: &Data<LemmyContext>,
+) -> LemmyResult<()> {
+  for community in to_follow {
+    let state = community_follower_state(community);
+    let mut form = CommunityFollowerForm::new(community.id, person.id, state);
+    form.is_multi_community_follow = Some(true);
+    CommunityActions::follow(&mut context.pool(), &form).await?;
```

Comment:
No need for the mut, just do 

```rust
CommunityFollowerForm {
  is_multi_community: Some(true),
  ..CommunityFollowerForm::new(community.id, person.id, state)
}
```

I'm not sure this DB field makes sense tho. I'll skip below.